presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -541,6 +543,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            Optional<Expression> argumentNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
```

Comment:
So I think you are referring to aliases. The arguments to `grouping` cannot refer to aliases in the select list. So for example this query should work (works):
```
select nationkey as b from nation group by nationkey order by grouping(nationkey)
```

But this one should fail:
```
select nationkey as b from nation group by nationkey order by grouping(nationkey)
```

Am I understanding correctly? If that's the case then I'm seeing something strange with one of my unit tests (after implementing your suggestion). The following query fails:

```
SELECT a, b, sum(c), grouping(a, b) + grouping(a)
FROM (VALUES ('h', 'j', 11), ('k', 'l', 7) ) AS t (a, b, c)
GROUP BY GROUPING SETS ( (a), (b))
ORDER BY grouping(b) ASC;
```

But when I alias the `b` column with something else and refer to it as `b` in the `grouping()`, it works. So the following runs fine:
```
SELECT a, b as t, sum(c), grouping(a, b) + grouping(a)
FROM (VALUES ('h', 'j', 11), ('k', 'l', 7) ) AS t (a, b, c)
GROUP BY GROUPING SETS ( (a), (b))
ORDER BY grouping(b) ASC;
```

> Still I think you should fix this issue in this PR to make the grouping feature complete.

Are you talking about 1) not allowing `grouping()` to reference output columns when in an order by or 2) adding support for `DereferenceExpressions` in `grouping()`?